platform/saivpp/vpplib/SwitchStateBaseFdb.cpp

```diff
@@ -1082,3 +1082,352 @@ sai_status_t SwitchStateBase::vpp_delete_bvi_interface(
+    br_port_id = attr_type->value.oid;
+    sai_object_type_t obj_type = objectTypeQuery(br_port_id);
+
+    if (obj_type != SAI_OBJECT_TYPE_BRIDGE_PORT)
+    {
+        SWSS_LOG_ERROR("SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID=%s expected to be PORT but is: %s",
+                sai_serialize_object_id(br_port_id).c_str(),
+                sai_serialize_object_type(obj_type).c_str());
+
+        return SAI_STATUS_FAILURE;
+    }
+
+    auto br_port_attrs = m_objectHash.at(SAI_OBJECT_TYPE_BRIDGE_PORT).at(sai_serialize_object_id(br_port_id));
+    auto meta = sai_metadata_get_attr_metadata(SAI_OBJECT_TYPE_BRIDGE_PORT, SAI_BRIDGE_PORT_ATTR_PORT_ID);
+    auto bp_attr = br_port_attrs[meta->attridname];
+    port_id = bp_attr->getAttr()->value.oid;
+    obj_type = objectTypeQuery(port_id);
+
+    if (obj_type != SAI_OBJECT_TYPE_PORT)
+    {
+        SWSS_LOG_NOTICE("SAI_BRIDGE_PORT_ATTR_PORT_ID=%s expected to be PORT but is: %s",
+                sai_serialize_object_id(port_id).c_str(),
+                sai_serialize_object_type(obj_type).c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    /* Need to extract the VLAN ID attached based on the Port_ID */
+    sai_attribute_t attr;
+    attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
+
+    sai_status_t get_status = get(SAI_OBJECT_TYPE_PORT, port_id, 1, &attr);
+
+    if (get_status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("failed to get port vlan id from port %s", sai_serialize_object_id(port_id).c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    uint32_t bd_id = attr.value.u16; /* bd_id is same as VLAN ID for .1Q bridge */
+
+    std::string ifname;
+    if (vpp_get_hwif_name(port_id, 0, ifname) == true) 
+    {
+        const char *hwif_name = ifname.c_str();
+        SWSS_LOG_NOTICE(" FDB_ENTRY on hwif_name %s ", hwif_name);
+        auto ret = l2fib_add_del(hwif_name, fdb_entry.mac_address, bd_id, is_add, is_static);
+        SWSS_LOG_NOTICE("FDB_Entry Added Success: %d", ret);
+
+    }
+    else
+    {
+        SWSS_LOG_ERROR("FDB: Invalid ObjectID for the hwif on this bridge");
+
+        return SAI_STATUS_FAILURE;
+    }
+
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t SwitchStateBase::vpp_fdbentry_del(
+        _In_ const std::string &serializedObjectId)
+{
+
+    SWSS_LOG_ENTER();
+           
+    sai_fdb_entry_t fdb_entry;
+    sai_deserialize_fdb_entry(serializedObjectId, fdb_entry);
+    
+    sai_object_id_t br_port_id;
+    sai_object_id_t port_id;
+    bool is_static = false;
+
+    sai_attribute_t attr_list[2];
+    /* Attribute#1 */
+	attr_list[0].id = SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID;
+    /* Attribute#2 */
+    attr_list[1].id = SAI_FDB_ENTRY_ATTR_TYPE;
+    
+	if (get(SAI_OBJECT_TYPE_FDB_ENTRY, serializedObjectId, 1, &attr_list[0]) == SAI_STATUS_SUCCESS)
+    {
+       if (SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID == attr_list[0].id)
+        {
+            br_port_id = attr_list[0].value.oid;
+        }
+        else
+        {
+            return SAI_STATUS_FAILURE;
```

Comment:
add debug error message before returning failure, as caller of this also do not have failure debug message